airflow/utils/log/file_task_handler.py

```diff
@@ -87,7 +149,14 @@ def close(self):
         if self.handler:
             self.handler.close()
 
-    def _render_filename(self, ti: TaskInstance, try_number: int) -> str:
+    def _render_filename(
+        self,
+        ti: TaskInstance | TaskInstanceKey,
+        try_number: int,
+        *,
+        is_trigger=False,
+    ) -> str:
+        try_number = try_number + 1 if is_trigger else try_number
```

Comment:
Ugh :(. I'm not sure there is a better way of doing this (other than entirely refactoring how try_number behaves which is a rather bigger issue than this PR)  but I don't like it

Reply:
yes, sigh